backend/controllers/brandController.js

```diff
@@ -82,13 +83,107 @@ exports.getProfile = async (req, res) => {
 };
 
 // Update brand's profile (requires authentication)
-exports.updateProfile = (req, res) => {
-  const brand = req.brand;
+exports.updateProfile = async (req, res) => {
+  const brandId = req.params.brandId;
+  const { name, email, password, category, location, website, description } = req.body;
+
+  const updatedFields = {
+    name: name || undefined,
+    email: email || undefined,
+    category: category || undefined,
+    location: location || undefined,
+    website: website || undefined,
+    description: description || undefined,
+  };
+
+  // Handle password separately
+  if (password) {
```

Comment:
I wonder if we should allow password update as well in update API or should we create a separate password reset API 'coz there is some extra checks needed for this no like the current password shouldn't match the new password etc

Reply:
Should I remove that password update from here?

Reply:
yes, please remove it. There should be separate APIs for reset and update password